lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -361,12 +361,56 @@ struct ExprVisitor {
     for (auto *operand : expr.operands()) {
       auto value = context.convertExpression(*operand);
       if (!value)
-        return {};
+        continue;
```

Comment:
Example:
```
  int a;
  assign a = {{0{32'b1}, 32'b0};
```
We just ingore `{0{32'b1}}`, then continue to handle `32'b0`, rather than directly `return{}`.
Please see IEEE 1800-2017§11.4.12.1 "Replication operator"
![image](https://github.com/llvm/circt/assets/105583479/a7130b8c-cba4-46ab-8748-1731f83784de)

But there is no test case, I'll add it when supporting `void` type.

Reply:
Can't we just use i0 constant for a zero replication constant?

Reply:
`i0` is permitted in FIRRTL, but an error will be thrown when existing in SV. I'll add the related error test cases for it.
Like this `logic l1 = {{0{1'b1}}};` Slang will throw an error: "concatenation cannot be empty".

Reply:
System Verilog doesn't support but SV dialect support i0. CIRCT removes i0 while ExportVerilog so I think it's OK to use i0.

Reply:
I don't think we can really treat the `void` type as a simple bit vector type. As far as I know, it's a pretty isolated type in the SV spec. We might have to add special treatment for replication operators nested in concatenations, to support zero-width repetitions. They can't exist as isolated expressions, so we probably shouldn't try to turn them into an expression at all. Maybe in the lowering for concatenations we can check if a concatenated value is a repetition, and then add special handling for the zero-width case.

If we switched the Moore dialect over to actually have a simple bit vector type, like `!moore.logic<16>`, we _could_ just lower zero repetitions to `!moore.logic<0>`. Maybe that's worth a shot.